yfinance/base.py

```diff
@@ -847,11 +847,18 @@ def get_news(self, proxy=None):
         self._news = data.get("news", [])
         return self._news
 
-    def get_earnings_dates(self, proxy=None) -> Optional[pd.DataFrame]:
-        if self._earnings_dates is not None:
-            return self._earnings_dates
+    def get_earnings_dates(self, limit=1000, proxy=None) -> Optional[pd.DataFrame]:
```

Comment:
Set `limit` default to 12, that will be enough data for most users - next 4 quarters and last 8 quarters.
Will need to think about documentation so users know how to get full history, i.e. `get_earnings_dates()` vs `earnings_dates` property.

Reply:
Fixed

Reply:
Adjust README too, currently says "# show all earnings dates"